src/transfer-manager.ts

```diff
@@ -331,12 +334,25 @@ export class TransferManager {
       start += chunkSize;
     }
 
-    return Promise.all(promises)
-      .then(results => {
-        return results.map(result => result.buffer) as DownloadResponse;
-      })
-      .finally(async () => {
-        await fileToWrite.close();
-      });
+    return new Promise((resolve, reject) => {
+      let results: DownloadResponse;
+      Promise.all(promises)
+        .then(data => {
+          results = data.map(result => result.buffer) as DownloadResponse;
+          if (options.validation === 'crc32c') {
+            return CRC32C.fromFile(filePath);
```

Comment:
This is a post-file download CRC32C validation (if enabled). It will occur if the user opts in and the file is downloaded as a single chunk or mutichunks